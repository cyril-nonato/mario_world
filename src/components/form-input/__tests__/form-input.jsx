import React from 'react'
import {render, cleanup, fireEvent} from '@testing-library/react'
import FormInput from '../form-input.component'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

describe('Name of the group', () => {

  afterEach(cleanup)
  
  const mockedProps = {
    placeholder: 'placeholder',
    name: 'name',
    type: 'type',
    label: 'label',
  }
  const onChange = jest.fn()
  let wrapper;
  beforeEach(() => {
    wrapper = render(<FormInput {...mockedProps} onChange={onChange} />)
  })
  test('Should render the correct props', () => {
    const {getByPlaceholderText, getByTestId} = wrapper;
  
    expect(getByPlaceholderText(mockedProps.placeholder)).toHaveAttribute('type', mockedProps.type);
    expect(getByPlaceholderText(mockedProps.placeholder)).toHaveAttribute('name', mockedProps.name);
    expect(getByTestId('label')).toHaveTextContent(mockedProps.label);
  });

  test('Should call onChange function', () => {
    const {getByTestId} = wrapper;

    fireEvent.change(getByTestId('input'), {target: {value: 'hello'}});
    expect(getByTestId('input').value).toBe('hello');
  });

  test('Should match the snapshot', () => {
    const {asFragment} = wrapper;

    expect(asFragment()).toMatchSnapshot();
  });
});