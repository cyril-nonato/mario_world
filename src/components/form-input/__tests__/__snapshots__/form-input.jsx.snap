// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group Should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  margin: 1.5rem 0;
}

.c1 {
  border: none;
  border-bottom: 2px solid #454545;
  padding: .5rem 0;
  font-size: 1.2rem;
  display: inline-block;
  width: inherit;
  font-family: inherit;
}

.c1:focus {
  outline: none;
}

.c1::-webkit-input-placeholder {
  color: #ccc;
  text-transform: capitalize;
  font-size: 1rem;
}

.c1::-moz-placeholder {
  color: #ccc;
  text-transform: capitalize;
  font-size: 1rem;
}

.c1:-ms-input-placeholder {
  color: #ccc;
  text-transform: capitalize;
  font-size: 1rem;
}

.c1::placeholder {
  color: #ccc;
  text-transform: capitalize;
  font-size: 1rem;
}

.c1:placeholder-shown ~ label {
  opacity: 0;
  visibility: hidden;
  -webkit-transform: translateY(-3rem);
  -ms-transform: translateY(-3rem);
  transform: translateY(-3rem);
}

.c2 {
  -webkit-transform: translateY(.5rem);
  -ms-transform: translateY(.5rem);
  transform: translateY(.5rem);
  -webkit-transition: all .3s;
  transition: all .3s;
  display: block;
  font-size: 1rem;
  text-transform: capitalize;
  font-weight: bolder;
}

<div
    class="c0"
  >
    <input
      class="c1"
      data-testid="input"
      name="name"
      placeholder="placeholder"
      required=""
      type="type"
    />
    <label
      class="c2"
      data-testid="label"
    >
      label
    </label>
  </div>
</DocumentFragment>
`;
